pandas/tests/groupby/test_categorical.py

```diff
@@ -453,6 +453,32 @@ def test_dataframe_categorical_with_nan(observed):
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.parametrize("ordered", [True, False])
+@pytest.mark.parametrize("observed", [True, False])
+@pytest.mark.parametrize("sort", [True, False])
+def test_dataframe_categorical_ordered_observed_sort(ordered, observed, sort):
+    # GH 25871: Fix groupby sorting on ordered Categoricals
+    # Build a dataframe with cat having one unobserved category ('AWOL'),
+    # and a Series with identical values
+    cat = pd.Categorical(['d', 'a', 'b', 'a', 'd', 'b'],
+                         categories=['a', 'b', 'AWOL', 'd'],
+                         ordered=ordered)
+    val = pd.Series(['d', 'a', 'b', 'a', 'd', 'b'])
+    df = pd.DataFrame({'cat': cat, 'val': val})
+
+    # aggregate on the Categorical
+    result = (df.groupby('cat', observed=observed, sort=sort)['val']
+                .aggregate('first'))
+
+    # If ordering works, we expect index labels equal to aggregation results,
+    # except for 'observed=False': index contains 'AWOL' and aggregation None
+    label = pd.Series(result.index.array, dtype='object')
+    aggr = pd.Series(result.array)
+    if not observed:
+        aggr[aggr.isna()] = 'AWOL'
+    tm.assert_equal(label, aggr)
+
+
```

Comment:
Can you also add the example from #25167?  Then we can close that one as well (since they were duplicate issues AFAICT).

Reply:
Will do, sure. I agree they seem duplicates.

Reply:
Done, checking now.
I added the sample as-is, without extended parameterisation, as that was originally submitted.
My new test covers all combinations of parameters, anyway.
Is that what you had in mind?